### ISSUE TITLE
derive: allow raw Params object as parameter

### ISSUE THREAD

**Issue by ascjones** [paritytech/jsonrpc#389]
Idea from https://github.com/paritytech/jsonrpc/issues/387#issuecomment-463104381

Allow a single raw `Params` object, skipping the automatic deserialization in the derive macro. e.g.
```
#[rpc(name = "double", rawParams)]
fn double(&self, params: Params) -> Result<u8> {
   let addArgs: AddArgs = params.parse()?;
}
```
> It always have to be single-parameter method (variable name does not matter) and then you can parse it either as array, object or both if you need it

**Comment by ebkalderon** [comment]
Is this issue being actively worked on? I might be willing to create a PR for this.

**Comment by ascjones** [comment]
No it's not currently being worked on, and yes please for a PR :smile:

**Comment by ebkalderon** [comment]
@ascjones Pull request is ready for review!

**Comment by ascjones** [comment]
Closed by #452
